Allow specific prune specific channel is a dict is defined (#68)
redhatci/ansible-collection-redhatci-ocp: roles/prune_catalog/README.md

# prune_catalog role

> A role to prune a File Base Catalogs (FBC), based on a list of required operators. The role will prune all the channels if  `pc_operators` is a list or a specific channel if it is a dictionary.

Reviewer: ```suggestion
A role to prune a File Base Catalog (FBC). The role creates a pruned catalog, leaving only the specified operators.
```

## Parameters

Name                   | Required | Default                                       | Description
---------------------- | -------- | --------------------------------------------- | ------------
pc_source_catalog      | Yes      |                                               | Source catalog to be pruned
pc_destination_catalog | Yes      |                                               | Catalog containing the required operators

> pc_operators           | Yes      |                                               | A List or dictionary of operators to include in the pruned catalog. See examples below

Reviewer: ```suggestion
pc_operators           | Yes      |                                               | The set of operators to keep in the pruned catalog. See examples below
```

    pc_opm_args: "--skip-tls-verify=true"
```

All channels for the listed operators are mirrored
```ShellSession
$ DOCKER_CONFIG=/home/<user>/.docker oc-mirror list operators --catalog <my-local-registry>:4443/pruned-catalog:latest --package compliance-operator
NAME                 DISPLAY NAME         DEFAULT CHANNEL
compliance-operator  Compliance Operator  stable

PACKAGE              CHANNEL      HEAD
compliance-operator  4.7          compliance-operator.v0.1.32
compliance-operator  release-0.1  compliance-operator.v0.1.61
compliance-operator  stable       compliance-operator.v1.3.1
```

> Prune a dictionary for specific channel

Reviewer: ```suggestion
Create a pruned catalog with specific channels
```

```yaml

> - name: "Prune a catalog specific with channels"

Reviewer: ```suggestion
- name: "Create a pruned catalog with specific channels"
```

  include_role:
    name: redhatci.ocp.prune_catalog
  vars:
    pc_source_catalog: "registry.redhat.io/redhat/redhat-operator-index:v4.13"
    pc_destination_catalog: "<my-local-registry>:4443/pruned-catalog:latest"
    pc_operators:
      compliance-operator:
        channel: stable
      file-integrity-operator:
        channel: stable
      cluster-logging:
        channel: stable
    pc_opm_args: "--skip-tls-verify=true"
```

> Only the specific channel for the listed operators are mirrored

Reviewer: ```suggestion
Only the specified channel of the operators are kept in the catalog
```

```ShellSession
$ DOCKER_CONFIG=/home/<user>/.docker oc-mirror list operators --catalog <my-local-registry>:4443/pruned-catalog:latest 
NAME                     DISPLAY NAME               DEFAULT CHANNEL
cluster-logging          Red Hat OpenShift Logging  stable
compliance-operator      Compliance Operator        stable
file-integrity-operator  File Integrity Operator    stable

$ DOCKER_CONFIG=/home/<user>/.docker oc-mirror list operators --catalog <my-local-registry>:4443/pruned-catalog:latest --package compliance-operator
NAME                 DISPLAY NAME         DEFAULT CHANNEL
compliance-operator  Compliance Operator  stable

PACKAGE              CHANNEL  HEAD
compliance-operator  stable   compliance-operator.v1.3.1
```

> Prune a dictionary for specific channel

Reviewer: ```suggestion
Another example of a pruned catalog with specific channels
```

```yaml

> - name: "Prune all and/or single channels"

Reviewer: ```suggestion
- name: "Create a pruned catalog with custom channels"
```

> - name: "Prune a all and/or single channels"

Reviewer 2: Just wonder if this is a typo `Prune a all ...` maybe it's just `Prune all ...`

Author: yes, it was a typo. Thank!

NAME                     DISPLAY NAME               DEFAULT CHANNEL
cluster-logging          Red Hat OpenShift Logging  stable-5.8
compliance-operator      Compliance Operator        stable
file-integrity-operator  File Integrity Operator    stable

$DOCKER_CONFIG=/home/<user>/.docker oc-mirror list operators --catalog <my-local-registry>:4443/pruned-catalog:latest --package file-integrity-operator
NAME                 DISPLAY NAME         DEFAULT CHANNEL
compliance-operator  Compliance Operator  stable

PACKAGE              CHANNEL      HEAD
compliance-operator  4.7          compliance-operator.v0.1.32
compliance-operator  release-0.1  compliance-operator.v0.1.61
compliance-operator  stable       compliance-operator.v1.3.1

$ DOCKER_CONFIG=/home/<user>/.docker oc-mirror list operators --catalog r<my-local-registry>:4443/pruned-catalog:latest --package file-integrity-operator
NAME                     DISPLAY NAME             DEFAULT CHANNEL
file-integrity-operator  File Integrity Operator  stable

PACKAGE                  CHANNEL  HEAD
file-integrity-operator  stable   file-integrity-operator.v1.3.3
```

### Catalog inspection

> Pruning an specific channels requires to be able to inspect the source index to identify the available channel to be pruned. oc-mirror can be used to perform the task. Eg:

Reviewer: ```suggestion
To keep specific channels, it is required to inspect the source catalog to identify the available channels. `oc-mirror` can be used to perform the task. Example:
```
